modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusStateUpdater.scala

```diff
@@ -345,6 +373,20 @@ object ConsensusStateUpdater {
       }
     }
 
+    private def checkFacilitatorsHashes[A](facilitatorsHash: A => Hash, ownFacilitatorsHash: Option[Hash])(
```

Comment:
1. Please rename this function to something like `warnIfForking`
1. Argument name `facilitatorHash` doesn't fit its type, it's not a `Hash` but a mapping function. Also it would be better if you would simply add `def facilitatorHash: Hash` to `PeerDeclaration` while a adding here a constraint on `A` to be `A <: PeerDeclaration`, then you wouldn't need this argument at all.
1. `ownFacilitatorHash` shouldn't be `Option`
1. The result of this function should be only `F[Unit]` as it logs stuff at most